internal/shim-sev/src/main.rs

```diff
@@ -38,31 +36,38 @@ mod start;
 pub mod syscall;
 pub mod usermode;
 
-use crate::addr::{ShimPhysUnencryptedAddr, ShimVirtAddr, SHIM_VIRT_OFFSET};
-use crate::attestation::SEV_SECRET;
-use crate::hostlib::BootInfo;
+use crate::attestation::SevSecret;
 use crate::pagetables::switch_sallyport_to_unencrypted;
 use crate::paging::SHIM_PAGETABLE;
 use crate::payload::PAYLOAD_VIRT_ADDR;
 use crate::print::{enable_printing, is_printing_enabled};
-use core::convert::TryFrom;
 use core::mem::size_of;
 use core::sync::atomic::{AtomicBool, AtomicU64, Ordering};
-use primordial::Address;
-use sallyport::Block;
 use spinning::RwLock;
 use x86_64::structures::paging::Translate;
 use x86_64::VirtAddr;
 
 static C_BIT_MASK: AtomicU64 = AtomicU64::new(0);
 
-static BOOT_INFO: RwLock<Option<BootInfo>> =
-    RwLock::<Option<BootInfo>>::const_new(spinning::RawRwLock::const_new(), None);
-
-static SHIM_HOSTCALL_PHYS_ADDR: RwLock<Option<usize>> =
-    RwLock::<Option<usize>>::const_new(spinning::RawRwLock::const_new(), None);
-
-static mut PAYLOAD_READY: AtomicBool = AtomicBool::new(false);
+static SEV_SECRET: RwLock<Option<SevSecret>> =
+    RwLock::<Option<SevSecret>>::const_new(spinning::RawRwLock::const_new(), None);
+
+static PAYLOAD_READY: AtomicBool = AtomicBool::new(false);
+
+extern "C" {
+    /// Extern
+    pub static _ENARX_SALLYPORT_START: u8;
+    /// Extern
+    pub static _ENARX_SALLYPORT_END: u8;
+    /// Extern
+    pub static _ENARX_MEM_START: u8;
+    /// Extern
+    pub static _ENARX_SHIM_START: u8;
+    /// Extern
+    pub static _ENARX_CODE_START: u8;
+    /// Extern
+    pub static _ENARX_CODE_END: u8;
```

Comment:
Why are these `u8` instead of their actual types?

Reply:
I don't want rust to have any assumptions about alignment.

Reply:
Don't you just force these assumptions on later code?

Reply:
Ok, done.